python_modules/dagster-graphql/dagster_graphql/implementation/execution/run_lifecycle.py

```diff
@@ -110,6 +112,7 @@ def create_valid_pipeline_run(
         status=DagsterRunStatus.NOT_STARTED,
         external_job_origin=external_pipeline.get_external_origin(),
         job_code_origin=external_pipeline.get_python_origin(),
+        asset_job_partitions_def=code_location.get_asset_job_partitions_def(external_pipeline),
```

Comment:
What's the size of the partitions def typically?  This can still be quite large if it's a static partitions def but small in all other cases?

For static partitions def, would we just want the serialization of the partition keys.  Maybe micro-optimization...

Reply:
If the point of partition ranges is just to help with write throughput, then should there be some threshold of how large the ranges are?

Reply:
> This can still be quite large if it's a static partitions def but small in all other cases?

Yes, this is true...

Are we worried about performance if the size of the partitions def is large? Between serializing a large execution plan versus having a large in-memory partitions def being passed around, I lean toward having the in-memory object

Reply:
> If the point of partition ranges is just to help with write throughput, then should there be some threshold of how large the ranges are?

I think its to minimize the # of created runs (maybe the same as what you mean by write throughput). 

I think it's sensible to limit the size of a range, though the range should be large enough to target all partitions of a reasonably-sized partitions def.

Maybe we need to first decide a limit on the size of partitions defs to enforce this?